### PR TITLE
Make distributed standard well work on SPE9.

### PR DESCRIPTION
**Pull request by blattms**
This an extension of #2954 that produces correct. It is rebased on current master. though (and includes the other PR). Anyway, to be safe, this should only be merged after #2954 and I will probably need to rebase it before.

Relevant changes are only the last 4 commits.

### CODE REVIEW

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
Like with the previous PR, the default is still to not allow wells to be distributed. Hence this should be safe to merge as long as jenkins agrees.

**Comment by blattms** [comment]
While my newer  version/branch is hanging this one seems to work fine for Norne with 1-32 procs. Undrafting.

**Comment by atgeirr** [comment]
Will review after rebase.

**Comment by blattms** [comment]
Concerning confusion: Welcome to the club.

I rebased this, fixed sequential build, and included a fix for a bug discovered on Norne.

**Comment by blattms** [comment]
jenkins built this serial parallel please

**Comment by atgeirr** [comment]
jenkins build this serial parallel please

File: opm/simulators/wells/ParallelWellInfo.hpp [hunk 1]
```diff
@@ -79,13 +82,88 @@ class CommunicateAbove
...
+            std::vector<int> displ(comm_.size(), 0);
+            using GlobalIndex = typename IndexSet::IndexPair::GlobalIndex;
+            using Pair = std::pair<GlobalIndex,Value>;
+            std::vector<Pair> my_pairs;
+            my_pairs.reserve(current_indices_.size());
+            for (const auto& pair: current_indices_)
+            {
+                if (pair.local().attribute() == owner)
+                {
+                    my_pairs.emplace_back(pair.global(), *(begin+(static_cast<const int&>(pair.local()))));
```
Comment (atgeirr) [inline opm/simulators/wells/ParallelWellInfo.hpp:131]: Looks correct, but why not just `begin[pair.local()]`?

Also on line 156.

**Review by atgeirr — APPROVED** [review]
Looks fine, just one question about some rather convoluted syntax used.

File: opm/simulators/wells/ParallelWellInfo.hpp (hunk 1 above)
Comment (blattms) [reply opm/simulators/wells/ParallelWellInfo.hpp]: I have no idea why I did not do that. Done now. Will merge once test has run.

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
jenkins is green, approved: in it goes.

**Pull request merged**